ws/room.go

```diff
@@ -48,48 +48,57 @@ func (r *Room) newSession(host, client xid.ID, rooms *Rooms) {
 		iceHost = []outgoing.ICEServer{{URLs: rooms.addresses("stun", false)}}
 		iceClient = []outgoing.ICEServer{{URLs: rooms.addresses("stun", false)}}
 	case ConnectionTURN:
-		hostPW := util.RandString(20)
-		clientPW := util.RandString(20)
-		hostName := id.String() + "host"
-		rooms.turnServer.Allow(hostName, hostPW, r.Users[host].Addr)
-		clientName := id.String() + "client"
-		rooms.turnServer.Allow(clientName, clientPW, r.Users[client].Addr)
+		clientAccount := &turn.Account{
+			Id: client,
+			IP: r.Users[host].Addr,
+		}
+		hostAccount := &turn.Account{
+			Id: host,
+			IP: r.Users[client].Addr,
+		}
+		// FIXME handles error
+		err := rooms.turnServer.AcceptAccounts(clientAccount, hostAccount)
+		if err != nil {
+			panic(err)
+		}
 		iceHost = []outgoing.ICEServer{{
 			URLs:       rooms.addresses("turn", true),
-			Credential: hostPW,
-			Username:   hostName,
+			Credential: hostAccount.Credential,
+			Username:   hostAccount.Username,
 		}}
 		iceClient = []outgoing.ICEServer{{
 			URLs:       rooms.addresses("turn", true),
-			Credential: clientPW,
-			Username:   clientName,
+			Credential: clientAccount.Credential,
+			Username:   clientAccount.Username,
 		}}
 
 	}
 	r.Users[host].Write <- outgoing.HostSession{Peer: client, ID: id, ICEServers: iceHost}
 	r.Users[client].Write <- outgoing.ClientSession{Peer: host, ID: id, ICEServers: iceClient}
 }
 
+// Rooms STUN/TURN addresses
+// prefix is `stun` or `turn`
 func (r *Rooms) addresses(prefix string, tcp bool) (result []string) {
 	if r.config.ExternalIPV4 != nil {
-		result = append(result, fmt.Sprintf("%s:%s:%s", prefix, r.config.ExternalIPV4.String(), r.turnServer.Port))
```

Comment:
Do I understand this correctly: If turn external is enabled, then external ip should point to the ip of the turn server? As there may not be on the same server?

Reply:
This is correct, the `ExternalIP` address is basically used not to point to where screego itself is, but where the STUN/TURN server is - previously as STUN/TURN was always part of screego the IP address would be the address of where screego is hosted, but with this PR it becomes just a special case of the full range of possibilities 🙂

Maybe we should keep `SCREEGO_SERVER_ADDRESS=0.0.0.0:5050` and `SCREEGO_TURN_ADDRESS=0.0.0.0:3478` dedicated to screego itself (where it should start listening), but `SCREEGO_EXTERNAL_IP` should really become "list of external IP **and port** of the STUN/TURN server", and so this `addresses` function would only use this parameter alone?

@athoune what do you think?

Reply:
handling multiple external turn server with their secret can be complex. Does Screego need a cluster of turn servers?

Reply:
ah no I meant a single TURN server (with single secret) but with possibly multiple combinations of IP and ports to reach it, for example coturn can listen on IPv4 and IPv6, for some scenarios it can actually have two IPv4, it can also run on multiple ports just because some ports can be blocked by firewall, so it would be nice to be able to specify all these combinations - but it's still a single TURN server 🙂

Reply:
`SCREEGO_EXTERNAL_IP` with multiple values should be implemented in another PR, IMHO.

Reply:
What is missing for approving this PR?